src/dune/install_rules.mli

```diff
@@ -3,6 +3,16 @@ open Stdune
 val gen_rules :
   Super_context.t -> dir:Path.Build.t -> Build_system.Subdir_set.t
 
-val init_meta_and_dune_package : Super_context.t -> dir:Path.Build.t -> unit
+(** Generate rules for [.dune-package] and [META.<package-name>] files in a
+    given super context and directory. Surprisingly, the current implementation
+    generates rules for {b all} directories of the project, which are filtered
```

Comment:
Where are they filtered out in build_system.ml? (Just for my own curiousity, I don't mean to block the PR).

Reply:
It's done to ensure determinism. If you generate rules for dir `a`, but it happens to generate rules in `b`, then you have a choice: 1) use them; 2) ignore them; 3) report error.

(1) has a problem with determinism: rules for (b) end up depending on whether or not we forced (a) yet;
(3) was the plan originally, but it was hard to make all rules work so we did (2) as a "temporary" measure. Maybe we can clean up things so we switch to (3) later, but it's not a trivial fix.

Reply:
Sorry, I read "why" where you said "where". Whoops. Let me try again...

Reply:
This line:
```
Rules.find rules_produced dir
```

effectively filters the rules to just the ones produced for `dir`.